### PR TITLE
Add a setuptools extension for generating Python protobufs

### PR DESCRIPTION
**Pull request by dlj-NaN**
This setuptools extension generates Python protobuf code from .proto files, so they
can be created as part of a normal Python build. The extension uses an installed
protoc binary, which can be explicitly specified if needed.

### CODE REVIEW

File: python/protobuf_distutils/setup.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        # These Python versions should match the protobuf package:
+        "Programming Language :: Python",
+        "Programming Language :: Python :: 2",
+        "Programming Language :: Python :: 2.7",
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.3",
+        "Programming Language :: Python :: 3.4",
+        "Programming Language :: Python :: 3.5",
+        "Programming Language :: Python :: 3.6",
+        "Programming Language :: Python :: 3.7",
```
Comment (bobhancock) [inline python/protobuf_distutils/setup.py:112]: We should also include 3.8.

Comment (dlj-NaN) [reply python/protobuf_distutils/setup.py:114]: Added.

File: python/protobuf_distutils/README.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+### Example setup.py configuration
+
+```python
+from setuptools import setup
+setup(
+    # ...
+    name='example_project',
+
+    # Require this package, but only for setup (not installation):
+    setup_requires=['protobuf_distutils'],
```
Comment (bobhancock) [inline python/protobuf_distutils/README.md:30]: The package installs as protobuf-disutils, with a hyphen not an underbar.

File: python/protobuf_distutils/README.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  will be placed under `output_dir` according to the relative source paths
+  under `source_dir`.
+
+  For example, the source file `${source_dir}/subdir/message.proto`
+  will be generated as the Python file `${output_dir}/subdir/message_pb2.py`.
+
+- `proto_files`:
+
+  Specific .proto files can be specified for generating code, instead of
+  searching for all .proto files under `source_path`.
```
Comment (bobhancock) [inline python/protobuf_distutils/README.md:90]: Must be a list of strings.

Comment (dlj-NaN) [reply python/protobuf_distutils/README.md:96]: Done.

**Pull request merged**